sky/utils/controller_utils.py

```diff
@@ -262,8 +262,10 @@ def _get_cloud_dependencies_installation_commands(
                 '  ARCH="amd64"; '
                 'fi && '
                 '(command -v kubectl &>/dev/null || '
-                '("https://dl.k8s.io/release/v1.31.6/bin/linux/$ARCH/kubectl" '
-                '&& sudo install -o root -g root -m 0755 '
+                '(curl -s -LO "https://dl.k8s.io/release/'
+                '$(curl -L -s https://dl.k8s.io/release/stable.txt)'
```

Comment:
Should we stick with the fixed version as our other places did?

Also, can we add a test for this?

Reply:
Our k8s base image has kubectl already baked into it, so a bad kubectl install attempt won't break anything. If you used a different base image for the controller (or ran on a different cloud, like @cg505 said), you'll run into this

Reply:
I think adding a test will be more difficult. Prefer to just merge this and follow up with cross-cloud tests later.